api/client/events.go

```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/brimdata/zed/api"
+	"github.com/brimdata/zed/zson"
+)
+
+type EventsClient struct {
+	resp        *Response
+	unmarshaler *zson.UnmarshalContext
+}
+
+func newEventsClient(resp *Response) *EventsClient {
+	unmarshaler := zson.NewUnmarshaler()
+	unmarshaler.Bind(
+		api.EventPool{},
+		api.EventBranch{},
+		api.EventBranchCommit{},
+	)
+	return &EventsClient{
+		resp:        resp,
+		unmarshaler: unmarshaler,
+	}
+}
+
+func (l *EventsClient) Recv() (string, interface{}, error) {
+	var kind, data string
+	_, err := fmt.Fscanf(l.resp.Body, "event: %s\ndata: %s\n\n\n", &kind, &data)
+	if err != nil {
+		if errors.Is(err, io.EOF) {
+			return "", nil, nil
+		}
+		return "", nil, err
+	}
+	var v interface{}
+	if err := l.unmarshaler.Unmarshal(data, &v); err != nil {
+		return "", nil, err
+	}
+	return kind, v, err
```

Comment:
Nit: Returning kind doesn't provide any additional information. Maybe drop it?

Reply:
You need kind because it distinguishes between pool-new and pool-deleted. I don't like that you need it but this is where we are.